csharp/src/Ice/Current.cs

```diff
@@ -1,187 +1,29 @@
 // Copyright (c) ZeroC, Inc.
 
-namespace Ice;
-
-public class Current
-{
-    public ObjectAdapter adapter;
-    public Connection con;
-    public Identity id;
-    public string facet;
-    public string operation;
-    public OperationMode mode;
-    public Dictionary<string, string> ctx;
-    public int requestId;
-    public EncodingVersion encoding;
-
-    public Current()
-    {
-        id = new Identity();
-        facet = "";
-        operation = "";
-        encoding = new EncodingVersion();
-    }
-
-    public Current(
-        ObjectAdapter adapter,
-        Connection con,
-        Identity id,
-        string facet,
-        string operation,
-        OperationMode mode,
-        Dictionary<string, string> ctx,
-        int requestId,
-        EncodingVersion encoding)
-    {
-        this.adapter = adapter;
-        this.con = con;
-        this.id = id;
-        this.facet = facet;
-        this.operation = operation;
-        this.mode = mode;
-        this.ctx = ctx;
-        this.requestId = requestId;
-        this.encoding = encoding;
-    }
+#nullable enable
 
-    public Current Clone() => (Current)MemberwiseClone();
-
-    public override int GetHashCode()
```

Comment:
It's not clear to me where we used these Equals / GetHashCode.

Is there is a good use-case for using Current as the key of a Dictionary?